src/main/flight/imu.c

```diff
@@ -407,6 +394,73 @@ static bool imuIsAccelerometerHealthy(void)
     return (81 < accMagnitude) && (accMagnitude < 121);
 }
 
+// Calculate the dcmKpGain to use. When armed, the gain is imuRuntimeConfig.dcm_kp * 1.0 scaling.
+// When disarmed after initial boot, the scaling is set to 10.0 for the first 20 seconds to speed up initial convergence.
+// After disarming we want to quickly reestablish convergence to deal with the attitude estimation being incorrect due to a crash.
+//   - wait for a 250ms period of low gyro activity to ensure the craft is not moving
+//   - use a large dcmKpGain value for 500ms to allow the attitude estimate to quickly converge
+//   - reset the gain back to the standard setting
+float imuCalcKpGain(timeUs_t currentTimeUs, bool useAcc, float gyroX, float gyroY, float gyroZ)
+{
+    static bool lastArmState = false;
+    static timeUs_t gyroQuietPeriodTimeEnd = 0;
+    static timeUs_t attitudeResetTimeEnd = 0;
+    static bool attitudeResetCompleted = false;
+    float ret;
+    bool attitudeResetActive = false;
+
+    const bool armState = ARMING_FLAG(ARMED);
+
+    if (!armState) {
+        if (!STATE(FIXED_WING)) {  // disable for fixed wing. probably wouldn't hurt anything but not needed or tested
+            if (lastArmState) {   // Just disarmed; start the gyro quiet period
+                gyroQuietPeriodTimeEnd = currentTimeUs + ATTITUDE_RESET_QUIET_TIME;
+                attitudeResetTimeEnd = 0;
+                attitudeResetCompleted = false;
+            }
+
+            // If gyro activity exceeds the threshold then restart the quiet period.
+            // Also, if the attitude reset has been complete and there is subsequent gyro activity then
```

Comment:
Retrigger threshold after reset may be considerably higher - 15 des/s will trigger just be picking quad up.

Probably not important, but may cause problems in configurator - user will mostly see fast mode

Also it would be almost equivalent to enabling fast mode always when disarmed ...

Reply:
Well part of the logic is to retrigger on picking up and righting the quad after a crash.  But the retrigger logic only is active after the first successful normal reset after disarm.  So most users wouldn't see this in the configurator under normal circumstances.  Only if they had enabled the safety switch on the motors tab and armed/disarmed while connected.

Reply:
There is no need to accelerate convergence when there is no gyro overflow (or extreme gyro noise) .. But it should not hurt

Reply:
Testing showed that even picking the quad up and flipping it over was enough to cause the attitude estimate to again diverge 30 degrees or more and then slowly converege once still.  So the logic to re-trigger was added in response to that.

Reply:
Ugly...It may be caused by wrong/inaccurate gyro scaling. But the error is huge, drift after loop in horizon mode must be terrible.
It should be investigated ...